File: Arrays/0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:

        hashmap = defaultdict(list)
        
        for i in strs:
            sorted_str = ''.join(sorted(i))
            hashmap[sorted_str].append(i)

        return list(hashmap.values())

    
        
        

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> hm = new HashMap<>();
        for(String s : strs){
            String[] split = s.split("");
            Arrays.sort(split);
            String joined = String.join("", split);
            hm.computeIfAbsent(joined, k -> new ArrayList<>()).add(s);
        }
        List<List<String>> out = new ArrayList<>(hm.values());
        return out;
    }
}